### ISSUE TITLE
Plugins does not load because of ToUpper's default culture behaviour

### ISSUE THREAD

**Issue by philophys** [dnGrep/dnGrep#429]
Today I noticed that dnGrep does not search in my PDF files. So I examined a little bit and found that it does not loads plugins. After a bit debugging with dnSpy I figured out that at line https://github.com/dnGrep/dnGrep/blob/4ad2b6d9d98029ebe03e6e037aa20a5bc9da65f0/dnGREP.Engines/GrepPlugin.cs#L101

`tempDllFilePath` becomes `null`. After a bit playing with dnGREP.Engines.*.plugin files I saw that if I set `Name` to `NAME` and `File` to `FILE` it load plugins succesfully. I guess, at this point, it is totaly clear that why plugin load fails. This bug occurs because of `ToUpper` uses default culture which is `tr-TR` in my environment. That is why at line
 https://github.com/dnGrep/dnGrep/blob/4ad2b6d9d98029ebe03e6e037aa20a5bc9da65f0/dnGREP.Engines/GrepPlugin.cs#L90

`File` becomes `FİLE`. This causes switch case at line https://github.com/dnGrep/dnGrep/blob/4ad2b6d9d98029ebe03e6e037aa20a5bc9da65f0/dnGREP.Engines/GrepPlugin.cs#L95

could not catch it.  Because `FİLE` is not `FILE`. So `dllFilePath` becomes `null` and exception occurs.

I don't think that it is necessary but here is error log file anyways:
```
<Error time="6.20.2020 12:56:49.175">
  <Message>Failed to initialize dnGREP.Engines.MsWord engine.</Message>
  <Exception>System.ArgumentNullException
   konum: dnGREP.Engines.GrepPlugin.LoadPluginSettings()
   konum: dnGREP.Engines.GrepEngineFactory.LoadPlugins()</Exception>
</Error>
<Error time="6.20.2020 12:56:49.175">
  <Message>Failed to initialize dnGREP.Engines.OpenXml engine.</Message>
  <Exception>System.ArgumentNullException
   konum: dnGREP.Engines.GrepPlugin.LoadPluginSettings()
   konum: dnGREP.Engines.GrepEngineFactory.LoadPlugins()</Exception>
</Error>
<Error time="6.20.2020 12:56:49.175">
  <Message>Failed to initialize dnGREP.Engines.Pdf engine.</Message>
  <Exception>System.ArgumentNullException
   konum: dnGREP.Engines.GrepPlugin.LoadPluginSettings()
   konum: dnGREP.Engines.GrepEngineFactory.LoadPlugins()</Exception>
</Error>
```

**Comment by doug24** [comment]
Thanks for the debugging!   One of those things that has been in the code forever.  There doesn't seem to be any reason to do the upper case conversion - the plugin config files are in source control with known casing.

**Comment by doug24** [comment]
Fixed in release 2.9.256
